scanner/indexer/indexer.go

```diff
@@ -142,7 +141,19 @@ func NewIndexer(ctx context.Context, cfg config.IndexerConfig) (Indexer, error)
 	}, nil
 }
 
+func castToConfig[T any](f func(cfg T)) func(o interface{}) error {
```

Comment:
nit: up to you, but `any` is the more "modern" version of `interface{}` and it interchangeable. Up to you if you want to keep this or change to `any`

Reply:
I think it makes sense to move to `any`.